src/NServiceBus.Core/ObjectBuilder/ContainerComponent.cs

```diff
@@ -0,0 +1,124 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using Container;
+    using ObjectBuilder;
+    using ObjectBuilder.Common;
+    using Settings;
+
+    class ContainerComponent
+    {
+        public ContainerComponent(SettingsHolder settings)
+        {
+            this.settings = settings;
+        }
+
+        public IConfigureComponents ContainerConfiguration { get; private set; }
+
+        public IBuilder Builder { get; private set; }
+
+        public void UseContainer<T>(Action<ContainerCustomizations> customizations = null) where T : ContainerDefinition, new()
+        {
+            customizations?.Invoke(new ContainerCustomizations(settings));
+
+            UseContainer(typeof(T));
+        }
+
+        public void UseContainer(Type definitionType)
+        {
+            UseContainer(definitionType.Construct<ContainerDefinition>().CreateContainer(settings));
+        }
+
+        public void UseContainer(IContainer container)
+        {
+            customContainer = container;
+        }
+
+        public void UseExternallyManagedBuilder(IBuilder builder)
```

Comment:
Should we be more defensive here, for example protecting against calling this if `InitializeWithInternallyManagedContainer` has been called already?

Reply:
I guess it wouldn't hurt (but see my comment about testing above)

Reply:
Replied above as well.

Reply:
On the fence unit tests tends to get in the way in my experience. Would we be ok with just adding the checks but not have a test for it?

Reply:
For, now I'm fine with leaving things as-is.